### PR TITLE
IMG Button not appearing if AutoStyle has ever been set in a layer

### PR DESCRIPTION
**Pull request by jesusbotella**
This PR fixes one of the problems listed in https://github.com/CartoDB/support/issues/1724, which is that IMG button in layer style does not appear sometimes.

It was caused by the autostyle check in FillColor input, that checked if `styleModel` had an AutoStyle id instead of checking if it was enabled in the layer. Seems like AutoStyle id is never deleted from `styleModel` even after disabling autoStyle.

### Acceptance
1. Add a point layer to a Builder map
2. Go to Style tab inside the points layer
3. Check that you can set an image marker to the points
4. Enable AutoStyle via a widget
5. Check that image button is not shown in style tab inside the layer
5. Disable AutoStyle
6. Check that image button is back there
More information in https://github.com/CartoDB/support/issues/1724

### CODE REVIEW

File: lib/assets/test/spec/builder/editor/style/style-form/style-form-default-model.spec.js [hunk 1]
```diff
@@ -22,7 +22,8 @@ describe('editor/style/style-form/style-form-default-model', function () {
       }
     });
 
-    spyOn(StyleFormDefaultModel.prototype, '_onChange').and.callThrough();
+    spyOn(StyleFormDefaultModel.prototype, '_onChange');
```
Comment (VictorVelarde) [inline lib/assets/test/spec/builder/editor/style/style-form/style-form-default-model.spec.js]: 👍

**Comment by jesusbotella** [comment]
I've been reviewing my own changes again and even though this solution works, it doesn't do what it should.

It works because `autoStyle` property is always undefined in layerModelDefinition, and that makes [this condition](https://github.com/CartoDB/cartodb/blob/645d52c1157d15be8299ddcb6f3e9a3dfb7c3ad5/lib/assets/javascripts/builder/editor/style/style-form/style-form-dictionary/fill-color.js#L81) true.

But it shouldn't always be `undefined` if autoStyle is applied. So I know (more or less) where the culprit can be but I couldn't come up with a fix yet.

**Comment by jesusbotella** [comment]
@VictorVelarde I introduced a few changes that improve the current autoStyle state, hopefully :)

**Comment by ramiroaznar** [comment]
@jesusbotella what is the status of this one?

**Comment by jesusbotella** [comment]
I think the client solved the issue in https://github.com/CartoDB/support/issues/1724 by deleting the layer and re-adding it but I couldn't come up with any solutions. The styles get corrupted at some point when autostyle is enabled and refresh the page or activate/deactivate autoStyle several times in a row.

**Comment by ramiroaznar** [comment]
@ivanmalagon I suppose this is not going to be implemented, right?

**Comment by ivanmalagon** [comment]
According to @jesusbotella comment, the customer fixed its issue and the "fix" was not working as expected so... no, this won't be fixed.

If another client runs into the same problem we have this as reference.

**Pull request closed without merging**